lua/nougat/cache/buffer.lua

```diff
@@ -159,6 +159,39 @@ local subscribe = {
       end)
     end
   end,
+  ["vim-gitgutter"] = function()
```

Comment:
How should we define the name of this function?

Here I'm using `"vim-gitgutter"` since it's the plugin's name: https://github.com/airblade/vim-gitgutter.

But I don't know how should I configure to use it as the data source of git status.

Reply:
This should go inside the existing `gitstatus` section: https://github.com/MunifTanjim/nougat.nvim/blob/700708b73c037e7800cbb27209247fdd1b6a122f/lua/nougat/cache/buffer.lua#L122C3-L122C12

First you need to detect if `vim-gitgutter` is installed here: https://github.com/MunifTanjim/nougat.nvim/blob/700708b73c037e7800cbb27209247fdd1b6a122f/lua/nougat/cache/buffer.lua#L126-L128

```lua
    elseif vim.fn.exists("*GitGutterGetHunkSummary") <= 0 then
      provider = "vim-gitgutter"
    end
```

Then add the rest of it below the existing `if provider == "gitsigns" then` codes:
https://github.com/MunifTanjim/nougat.nvim/blob/700708b73c037e7800cbb27209247fdd1b6a122f/lua/nougat/cache/buffer.lua#L134-L160C7

```lua
    elseif provider == "vim-gitgutter" then
      on_event(...)
    end
```

Reply:
After you make these changes, no extra configuration is needed.

Reply:
update: fixed the 'gitstatus' subscribe for 'vim-gitgutter'.

But it still not working on my laptop.

Here's the configure:

```lua

local function git_branch()
    if vim.g.loaded_gitbranch == nil or vim.g.loaded_gitbranch <= 0 then
        return ""
    end
    local name = vim.fn["gitbranch#name"]()
    return name or ""
end

-- git branch
stl:add_item(Item({
    hl = { bg = color.magenta, fg = color.bg },
    sep_right = sep.right_upper_triangle_solid(true),
    prefix = "  ",
    suffix = " ",
    type = "lua_expr",
    content = function(ctx)
        local branch = git_branch()
        return branch or ""
        -- if string.len(branch) == 0 then
        --     return ""
        -- end
        -- local changes = git_diff()
        -- if string.len(changes) == 0 then
        --     return branch
        -- else
        --     return branch .. " " .. changes
        -- end
    end,
}))

-- -- git changes
stl:add_item(nut.git.status.create({
    hl = { fg = color.bg },
    content = {
        nut.git.status.count("added", {
            hl = { bg = color.green },
            prefix = "+",
            sep_right = sep.right_upper_triangle_solid(true),
        }),
        nut.git.status.count("changed", {
            hl = { bg = color.yellow },
            prefix = "~",
            sep_right = sep.right_upper_triangle_solid(true),
        }),
        nut.git.status.count("removed", {
            hl = { bg = color.red },
            prefix = "-",
            sep_right = sep.right_upper_triangle_solid(true),
        }),
    },
}))
```

It shows:

<img width="1234" alt="image" src="https://github.com/MunifTanjim/nougat.nvim/assets/6496887/b8c18aa4-b243-4e39-ab81-c817e5697a40">

Reply:
sorry, I forgot to add 'vim-gitgutter' as 'nougat.nvim' dependency, so it's actually not loaded. now it's working:

<img width="789" alt="image" src="https://github.com/MunifTanjim/nougat.nvim/assets/6496887/dc2edac3-0611-48a9-8964-06968c1cf7c7">

Reply:
hope more and more colorschemes support nougat, because I'm using [colorbox.nvim](https://github.com/linrongbin16/colorbox.nvim), it contains 41 maintained and most github stars colorschemes.

Reply:
> sorry, I forgot to add 'vim-gitgutter' as 'nougat.nvim' dependency, so it's actually not loaded. now it's working.

Yep, `vim-gitgutter` needs to be loaded before `nougat.nvim` is executed.